example/Example.res

```diff
@@ -15,6 +15,10 @@ module Form = {
       (),
     )
 
+    let hobbies: array<Values.Hobby.t> = Hooks.WatchValues.use(
+      ~option=Hooks.WatchValues.option(~control, ~name="hobbies", ()),
+    )
```

Comment:
I wonder @infothien , is this value always serializable (does it contain only Jsln values)?
If so we can use `Js.Json.t` as the returned value and ReCode to decode it 😁

Reply:
it could return everything based on the type of that field in the Form
```
type Form = {
   age: number
   name: string
   hobbies: array<Hobby>
}

const currentName = useWatch(control, "name") // it will be string: "thien"
const currentAge = useWatch(control, "age") // it will be number: "26"
const hobbies = useWatch(control, "hobbies") // it will be number: [{name: "game"}, {name: "exercise"]

```

But @gaku-sei -san, may be we dont really need this one for now. It might be use full when you want to display conditional fields based on currentValue of other field 

But for solving the problem we got right now in ATS, #4 this pr will work :D

Reply:
We don't need it now, for sure.

But it seemed pretty simple to add it so we can probably implement it using Js.Json.t so it remains flexible 😄

Reply:
sorry @gaku-sei -san, sorry I am not really sure what I should change here. As what I understand, useWatch wont return Json things but instead of the value type of the field in the form.
So I binded it in the hook file like this one. I might need to use this one in the end to fix the bug here https://scoville.atlassian.net/jira/software/projects/ATS/boards/107?selectedIssue=ATS-59 but not feel so comfortable with re-decode yet. So could you give a hand or more detail on it 🙇 
<img width="804" alt="Screenshot_2021_05_24_18_22" src="https://user-images.githubusercontent.com/30365498/119326619-0abe0c00-bcbd-11eb-9f28-76157cd4a8a8.png">